gc/base/MemoryPoolLargeObjects.cpp

```diff
@@ -948,11 +957,12 @@ MM_MemoryPoolLargeObjects::contractWithRange(MM_EnvironmentBase* env, uintptr_t
 		/* No LOA so just remove memory from SOA and we are done */
 		_memoryPoolSmallObjects->contractWithRange(env, contractSize, lowAddress, highAddress);
 	} else {
-
-		assume0(lowAddress >= _currentLOABase);
-
-		/* First remove memory from LOA */
-		_memoryPoolLargeObjects->contractWithRange(env, contractSize, lowAddress, highAddress);
+		if (lowAddress >= _currentLOABase) {
+			_memoryPoolLargeObjects->contractWithRange(env, contractSize, lowAddress, highAddress);
+		} else {
```

Comment:
we should probably assert that LOA is empty

(getCurrentLOASize() == getApproximateFreeLOAMemorySize())